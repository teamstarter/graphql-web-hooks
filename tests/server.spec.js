const request = require('supertest')
const { Op, Sequelize } = require('sequelize')
const path = require('path')

const {
  migrateDatabase,
  seedDatabase,
  getNewServer,
  models,
  closeEverything,
  deleteTables,
  resetDatabase,
} = require('./test-database.js')
const { removeTimestamps } = require('./tools.js')
const { getNewClient, callWebhook, getCallWebhook } = require('../lib/index')

// This is the maximum amount of time the band of test can run before timing-out
jest.setTimeout(600000)

let server = null
const client = getNewClient(
  `http://localhost:${process.env.PORT || 8080}/graphql`
)

const wait = (time) => new Promise((resolve) => setTimeout(resolve, time))

const webhook = (variables) => ({
  query: `query webhook($where: SequelizeJSON) {
    webhook(where: $where) {
      url
      headers {
        key
        value
      }
    }
  }`,
  variables,
  operationName: null,
})

const webhookCreate = (variables) => ({
  query: `mutation webhookCreate($webhook: webhookInput!) {
    webhookCreate(webhook: $webhook) {
      url
      securityMetadata
      headers {
        key
        value
      }
    }
  }`,
  variables,
  operationName: null,
})

const webhookDelete = (variables) => ({
  query: `mutation webhookDelete($id: Int!) {
    webhookDelete(id: $id)
  }`,
  variables,
  operationName: null,
})

/**
 * Starting the tests
 */
describe('Test server options', () => {
  beforeAll(async () => {
    await migrateDatabase()
    await seedDatabase()
    
  })

  beforeEach(async () => {
    await resetDatabase()
  })

  afterEach(async () => {
    await deleteTables()
  })

  afterAll(async () => {
    await closeEverything(server, models)
  })

  it('Check the server can be initialize with a sequelize connection already connected.', async () => {
    var dbConfig = require(path.join(__dirname, '/sqliteTestConfig.js')).test
    const sequelize = new Sequelize(dbConfig)
    server = await getNewServer(sequelize)

    const response = await request(server)
      .post('/graphql')
      .set('userId', 1)
      .send(webhook({}))

    expect(response.body.errors).toBeUndefined()
    expect(response.body.data).toMatchSnapshot()
  })
})
