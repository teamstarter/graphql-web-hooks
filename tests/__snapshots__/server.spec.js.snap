// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test server options Check the server can be initialize with a sequelize connection already connected. 1`] = `
Object {
  "webhook": Array [
    Object {
      "headers": Array [
        Object {
          "key": "header-key-1",
          "value": "header-value-1",
        },
        Object {
          "key": "header-key-2",
          "value": "header-value-2",
        },
      ],
      "url": "https://www.google.com",
    },
    Object {
      "headers": Array [],
      "url": "https://www.google.com",
    },
  ],
}
`;
